### ISSUE TITLE
Invalid column index.

### ISSUE THREAD

**Issue by jack128** [linq2db/linq2db#75]
Unit test     

``` C#
   [Test, DataContextSource]
        public void InvalidColumnIndex(string context)
        {
            using (var db = GetDataContext(context))
            {
                var childs = db.Child.Select(c => new 
                {
                    Child = c,
                    HasChildren = db.Child.Any(c2 => c2.ParentID == c.ChildID)
                });
                childs =
                    from child in childs
                    join parent in db.Parent on child.Child.ParentID equals parent.ParentID
                    where parent.Value1 == 1
                    select child;
                Assert.That(childs.ToList().Count, Is.GreaterThan(0));
            }
        }
```

generated sql

``` sql
SELECT
    child.ParentID as ParentID1,
    child.ChildID as ChildID1 -- "HasChildren" is not selected in query
FROM
    (
        SELECT
            c.ParentID,
            c.ChildID,
            CASE WHEN EXISTS(
                SELECT
                    *
                FROM
                    Child c2
                WHERE
                    c2.ParentID = c.ChildID
            ) THEN 1 ELSE 0 END as c1
        FROM
            Child c
    ) child
        INNER JOIN Parent parent ON child.ParentID = parent.ParentID
WHERE
    parent.Value1 = 1
```

**Comment by jogibear9988** [comment]
#479 is a test for this

**Comment by jogibear9988** [comment]
maybe anyone has a clue how to fix?

**Comment by jogibear9988** [comment]
close?

**Comment by ili** [comment]
Yes :) 
and released in [1.7.6](https://github.com/linq2db/linq2db/releases/tag/Release.1.7.6)
